### PR TITLE
Dashboard

### LINKED ISSUE #13: Dashboard
As an authenticated user,
When I visit '/dashboard'
I should see: 
- [x] 'Welcome `<username>`!' at the top of page
- [x] #9  A button to Discover Movies
- [x] #10 A friends section
- [x] #11 A viewing parties section

### PR DESCRIPTION
**Pull request by jakeheft**
As an authenticated user,
When I visit '/dashboard'
I should see:

 'Welcome <username>!' at the top of page
 #9 A button to Discover Movies
 #10 A friends section
 #11 A viewing parties section

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,8 +7,8 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
-  # def require_user
-  #   render file: '/public/401' unless current_user
-  # end
+  def require_user
+    render file: '/public/401' unless current_user
+  end
```
Comment (BJSherman80) [inline app/controllers/application_controller.rb:13]: nice and simple, looks great

File: app/controllers/friendship_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def create
+    friend = User.find_by(email: params[:email])
+    friendship = Friendship.new(user: current_user, friend: friend)
+    if current_user.email == params[:email]
+      flash[:failure] = "That's your own email, silly!"
+    elsif User.find_by(email: params[:email])
+      friendship.save
+      flash[:success] = "#{friend.name} is now your friend"
+    else
+      flash[:failure] = "There aren't any users with that email"
```
Comment (BJSherman80) [inline app/controllers/friendship_controller.rb:11]: nice sad path

File: app/controllers/party_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class PartyController < ApplicationController
+  def new; end
+
+  def create
+    require "pry"; binding.pry
```
Comment (dcoleman21) [inline app/controllers/party_controller.rb:5]: Let's pull this out.

**Review by BJSherman80 — APPROVED** [review]

**Review by dcoleman21 — APPROVED** [review]
looks good

**Pull request merged**